intake/source/base.py

```diff
@@ -70,12 +70,19 @@ def __setstate__(self, state):
     def __init__(self, metadata=None):
         # default data
         self.metadata = metadata or {}
+        if isinstance(self.metadata, dict):
+            self.cache = make_caches(self.name, self.metadata.get('cache'))
         self.datashape = None
         self.dtype = None
         self.shape = None
         self.npartitions = 0
         self._schema = None
 
+    def _get_cache(self, urlpath):
+        if len(self.cache) == 0:
+            return [urlpath]
+        return [c.load(urlpath) for c in self.cache]
```

Comment:
That's right.  We don't have any tests for that yet, since I don't have a data source that really needs multiple caching strategies.  I mostly wanted to sketch out how this would look.  We can certainly document how this could be done.

Reply:
If we support it, we should demonstrate it and test it :)